### PR TITLE
Compilation Improvements RC4.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: regression_tests/type_system/struct_fields_reflection.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+namespace type_test {
+
+#include "include/struct_fields.h"
+
+}  // namespace type_test
+
+TEST(TypeTest, StructFields) {
+  using namespace type_test;
+  using current::reflection::StructSchema;
+  using current::reflection::Language;
```
Comment (mzhurovich) [inline regression_tests/type_system/struct_fields_reflection.cc]: We don't need these two usings here, right?

File: regression_tests/type_system/test.sh [hunk 2]
```diff
@@ -3,6 +3,7 @@ COUNT_LIST=${*:-10 25 100 500}
 TEST_LIST=(
     'current_struct'
```
Comment (mzhurovich) [inline regression_tests/type_system/test.sh]: `current_structs` -> `current_structs_reflection`?

File: regression_tests/type_system/struct_fields_reflection.cc (hunk 1 above)
Comment (dkorolev) [inline regression_tests/type_system/struct_fields_reflection.cc]: :1: 5a86a94

File: regression_tests/type_system/test.sh (hunk 2 above)
Comment (dkorolev) [inline regression_tests/type_system/test.sh]: :+1: 5a86a94

File: TypeSystem/variant.h [hunk 3]
```diff
@@ -43,20 +52,91 @@ namespace current {
...
+  RuntimeTypeListHelpersImpl() : types_(PopulateTypes()) {}
+  template <typename T>
+  void AssertContains() const {
+    if (types_.find(std::type_index(typeid(T))) == types_.end()) {
+      std::cerr << "Variant type mismatch.\n\tType: " << reflection::CurrentTypeNameAsConstCharPtr<T>()
+                << "\nIs not among:";
+      for (const auto& type : types_) {
+        std::cerr << ' ' << type.second;
+      }
+      std::cerr << '\n';
```
Comment (mzhurovich) [inline TypeSystem/variant.h]: Let's remove or `ifdef` this `cerr` stuff?

Comment (dkorolev) [inline TypeSystem/variant.h]: I thought of it -- the only downside it extra output of the test, and the upside is simplified debugging.

Perhaps `#ifndef NDEBUG` ?

**Pull request merged**